src/lsm/table_mutable.zig

```diff
@@ -127,23 +254,19 @@ pub fn TableMutableType(comptime Table: type, comptime tree_name: [:0]const u8)
             table.value_count_worst_case += 1;
             switch (usage) {
                 .secondary_index => {
-                    const existing = table.values.fetchRemove(value.*);
-                    if (existing) |kv| {
+                    if (table.values.fetch_remove(value)) |existing| {
                         // The previous operation on this key then it must have been a put.
                         // The put and remove cancel out.
-                        assert(!tombstone(&kv.key));
+                        assert(!tombstone(existing));
                     } else {
                         // If the put is already on-disk, then we need to follow it with a tombstone.
                         // The put and the tombstone may cancel each other out later during compaction.
-                        table.values.putAssumeCapacityNoClobber(tombstone_from_key(key_from_value(value)), {});
+                        table.values.insert_no_clobber(tombstone_from_key(key_from_value(value)));
                     }
                 },
                 .general => {
-                    // If the key is already present in the hash map, the old key will not be overwritten
-                    // by the new one if using e.g. putAssumeCapacity(). Instead we must use the lower
-                    // level getOrPut() API and manually overwrite the old key.
-                    const upsert = table.values.getOrPutAssumeCapacity(value.*);
-                    upsert.key_ptr.* = tombstone_from_key(key_from_value(value));
+                    // Overwrite the old key and value if any.
+                    table.values.upsert(value).* = tombstone_from_key(key_from_value(value));
```

Comment:
Nice API improvement here with `upsert()`.

Reply:
We could improve the comment, since this not only overwrites the old key if any, but will otherwise also insert.